.github/workflows/format.yml

```diff
@@ -1,6 +1,6 @@
 name: Scalafmt
 
-permissions: read-all
+permissions: {}
```

Comment:
I don't think read-all is needed. `{}` disables the jobs access to repository secrets - and this should be the go-to setting unless a job needs to read secrets

Reply:
https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

Reply:
Are you sure this is what you intend? `read-all` is more strict than `{}`, not less.

Reply:
Are you sure this is what you intend? `read-all` is more strict than `{}`, not less.

Reply:
Thanks for clarifying, had a misunderstanding then

Reply:
To be frank, the Github docs are not well written - leaving plenty of ambiguity.

I'm not sure why not setting permissions at all in the yml does not default to having the lowest available permissions.

A few months ago, there were people popping PRs on dozens of repos to set `permissions: {}` but honestly, they could have been wrong.